pallene/to_ir.lua

```diff
@@ -257,6 +247,40 @@ function ToIR:convert_toplevel(prog_ast)
             -- skip
         end
     end
+
+    -- Initialize the module exports as a lua table
+    local exports_type  = types.T.Table({})
+    self.module.loc_id_of_exports = ir.add_local(self.func, "$exports", exports_type)
+    table.insert(cmds, ir.Cmd.NewTable(self.func.loc,  self.module.loc_id_of_exports, ir.Value.Integer(n_exports)))
+    table.insert(cmds, ir.Cmd.CheckGC())
+
+    -- export the functions
+    for _, f_id in ipairs(self.module.exported_functions) do
+        local func   = self.module.functions[f_id]
+        local loc_id = assert(self.loc_id_of_exported_func[f_id])
+
+        local tv = ir.Value.LocalVar( self.module.loc_id_of_exports)
+        local kv = ir.Value.String(func.name)
+        local fv = ir.Value.LocalVar(loc_id)
+        local src_typ = func.typ
+        table.insert(cmds, ir.Cmd.SetTable(func.loc, src_typ, tv, kv, fv))
+    end
+
+    -- export module variables
+    for _, loc_id in ipairs(self.module.exported_globals) do
+        local var  = self.func.vars[loc_id]
+        local name = assert(var.name)
+
+        local tv = ir.Value.LocalVar(self.module.loc_id_of_exports)
+        local kv = ir.Value.String(name)
+        local fv = ir.Value.LocalVar(loc_id)
+        local src_typ = var.typ
+        table.insert(cmds, ir.Cmd.SetTable(var.loc, src_typ, tv, kv, fv))
+    end
```

Comment:
These two loops look very similar. Maybe we could put the common stuff in a helper function?